packages/admin-next/dashboard/src/v2-routes/orders/order-create-fulfillment/components/order-create-fulfillment-form/order-create-fulfillment-form.tsx

```diff
@@ -0,0 +1,289 @@
+
+import {
+  RouteFocusModal,
+  useRouteModal,
+} from "../../../../../components/route-modal"
+import { CreateFulfillmentSchema } from "./constants"
+import { Form } from "../../../../../components/common/form"
+import { OrderCreateFulfillmentItem } from "./order-create-fulfillment-item"
+import { getFulfillableQuantity } from "../../../../../lib/order-item"
+import { useCreateFulfillment } from "../../../../../hooks/api/fulfillment"
+import { useStockLocations } from "../../../../../hooks/api/stock-locations"
+import { useFulfillmentProviders } from "../../../../../hooks/api/fulfillment-providers"
+import { cleanNonValues, pick } from "../../../../../lib/common"
+
+type OrderCreateFulfillmentFormProps = {
+  order: OrderDTO
+}
+
+export function OrderCreateFulfillmentForm({
+  order,
+}: OrderCreateFulfillmentFormProps) {
+  const { t } = useTranslation()
+  const { handleSuccess } = useRouteModal()
+
+  const { mutateAsync: createOrderFulfillment, isLoading: isMutating } =
+    useCreateFulfillment()
+
+  const { fulfillment_providers } = useFulfillmentProviders({
+    region_id: order.region_id,
+  })
+
+  const [fulfillableItems, setFulfillableItems] = useState(() =>
+    order.items.filter((item) => getFulfillableQuantity(item) > 0)
+  )
+
+  const form = useForm<zod.infer<typeof CreateFulfillmentSchema>>({
+    defaultValues: {
+      quantity: fulfillableItems.reduce((acc, item) => {
+        acc[item.id] = getFulfillableQuantity(item)
+        return acc
+      }, {} as Record<string, number>),
+      // send_notification: !order.no_notification,
+    },
+    resolver: zodResolver(CreateFulfillmentSchema),
+  })
+
+  const { stock_locations = [] } = useStockLocations()
+
+  const handleSubmit = form.handleSubmit(async (data) => {
+    try {
+      await createOrderFulfillment({
+        location_id: data.location_id,
+        /**
+         * TODO: send notification flag
+         */
+        // no_notification: !data.send_notification,
+        delivery_address: cleanNonValues(
+          pick(order.shipping_address, [
+            "first_name",
+            "last_name",
+            "phone",
+            "company",
+            "address_1",
+            "address_2",
+            "city",
+            "country_code",
+            "province",
+            "postal_code",
+            "metadata",
+          ])
+        ), // TODO: this should be pulled from order in the workflow
+        provider_id: fulfillment_providers[0]?.id,
+        items: Object.entries(data.quantity)
+          .filter(([, value]) => !!value)
+          .map(([item_id, quantity]) => {
+            const item = order.items.find((i) => i.id === item_id)
+
+            return {
+              quantity,
+              line_item_id: item_id,
+              title: item.title,
+              barcode: item.variant.barcode || "",
+              sku: item.variant_sku || "",
+            }
+          }),
+        // TODO: should be optional in the enpoint?
```

Comment:
Yeah, we should allow these to be manually passed. If you're using a provider, it will be responsible for the labels, but in some cases you would like to manually fulfill orders.